src/game/Spell.cpp

```diff
@@ -1958,9 +1958,40 @@ void Spell::SetTargetMap(SpellEffectIndex effIndex, uint32 targetMode, UnitList&
         }
         case TARGET_SINGLE_FRIEND:
         case TARGET_SINGLE_FRIEND_2:
-            if (m_targets.getUnitTarget())
-                targetUnitMap.push_back(m_targets.getUnitTarget());
+        {
+            if (Unit* unitTarget = m_targets.getUnitTarget())
+            {
+                if (unitTarget->IsFriendlyTo(m_caster))
+                    targetUnitMap.push_back(unitTarget);
+                else
+                {
```

Comment:
There's nothing wrong with it, you're right. It's just yet another design choise.... Several lines of additional code... in what is already somewhat lengthy files... :p